XYZDrivers/src/main/java/com/xyzdrivers/controllers/ClaimStatusController.java

```diff
@@ -0,0 +1,113 @@
+package com.xyzdrivers.controllers;
+
+import com.xyzdrivers.models.Claim;
+import com.xyzdrivers.repositories.ClaimsRepo;
+import com.xyzdrivers.repositories.RepositoryException;
+import com.xyzdrivers.services.ResponseService;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ *
+ * @author arthur
+ */
+public class ClaimStatusController extends HttpServlet {
+    
+    @Inject
+    ClaimsRepo claimsRepo;
+
+    /**
+     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
+     * methods.
+     *
+     * @param request servlet request
+     * @param response servlet response
+     * @throws ServletException if a servlet-specific error occurs
+     * @throws IOException if an I/O error occurs
+     */
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        
+        // Get the http request params
+        String status = request.getParameter("status");
+        Integer id = Integer.parseInt(request.getParameter("claim_id"));
+        
+        // Sanity check to ensure required parameters were required, if not respond with failure
+        if (status == null || status.equals("")) {
+            ResponseService.fail(request, response, "Failure. Please submit a status and claim id", "admin");
```

Comment:
Damn, I actually implemented a base controller that does something like this in #24. 
Your way is a bit cleaner - if mine gets merged first can you modify AdminController and PaymentController to use your way?

Reply:
sure